framework/src/actions/MaterialOutputAction.C

```diff
@@ -93,7 +89,8 @@ MaterialOutputAction::buildMaterialOutputObjects(FEProblemBase * problem_ptr)
   {
```

Comment:
As far as I can tell from the problem @sapitts is having, this loop is the actual problem. We need to loop over actual Output objects. She is adding Output objects via FEProblemBase::addOutput, so they don't show up in this action-based list, thus directing VPP objects to the output fails.

Reply:
As we discussed any Action may satisfy the add_outputs task, and if they register with that task, they will be returned with the rest of the built-in Actions. This should be working.